app/models/stat.rb

```diff
@@ -0,0 +1,66 @@
+# frozen_string_literal: true
+
+class Stat
+  def initialize(name, show_in_ui: false, expose_via_api: false, &block)
+    @name = name
+    @show_in_ui = show_in_ui
+    @expose_via_api = expose_via_api
+    @block = block
+  end
+
+  attr_reader :name, :expose_via_api, :show_in_ui
+
+  def calculate
+    begin
+      return @block.call.transform_keys { |key| build_key(key) }
+    rescue StandardError => err
+      Discourse.warn_exception(
+        err,
+        message: "Unexpected error when collecting #{@name} About stats.",
+      )
+    end
```

Comment:
what would be a possible error here?

Reply:
Honestly I don't know, I was refactoring the existing implementation, this part was just moved around. I think it's worth leaving it as is to avoid increasing the scope of the refactoring.